haystack_experimental/dataclasses/tool.py

```diff
@@ -106,7 +106,7 @@ def from_dict(cls, data: Dict[str, Any]) -> "Tool":
         return cls(**data)
 
     @classmethod
-    def from_function(cls, function: Callable, docstring_as_desc: bool = True) -> "Tool":
```

Comment:
I've decided to remove `docstring_as_desc` since the new parameters allow for more configurability.
We can do that because we are in experimental.